### PR TITLE
[IIDM v1.4.1] Removal of parameters on defaulted/deleted special functions (#220)

### PR DESCRIPTION
**Pull request by sebalaig**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#220 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: include/powsybl/stdcxx/reference_wrapper.hpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ class reference_wrapper {
 
     reference_wrapper(const reference_wrapper&) = default;
 
-    reference_wrapper(T&& reference) noexcept = delete;
+    reference_wrapper(T&&) noexcept = delete;
```
Comment (mathbagu) [inline include/powsybl/stdcxx/reference_wrapper.hpp:32]: Maybe, to avoid conflicts, you should revert this change and do it in the PR relative to reference_wrapper

**Review by mathbagu — APPROVED** [review]

File: include/powsybl/stdcxx/reference_wrapper.hpp (hunk 1 above)
Comment (sebalaig) [reply include/powsybl/stdcxx/reference_wrapper.hpp]: Agree, it is fixed in PR #266.
